bash-preexec.sh

```diff
@@ -284,7 +291,10 @@ __bp_install() {
     trap '__bp_preexec_invoke_exec "$_"' DEBUG
 
     # Preserve any prior DEBUG trap as a preexec function
-    local prior_trap=$(sed "s/[^']*'\(.*\)'[^']*/\1/" <<<"${__bp_trap_string:-}")
+    local prior_trap
+    # we can't easily do this with variable expansion. Leaving as sed command.
+	# shellcheck disable=SC2001
+	prior_trap=$(sed "s/[^']*'\(.*\)'[^']*/\1/" <<<"${__bp_trap_string:-}")
```

Comment:
If we want to eliminate the use of `sed`, we may use the regular expression through `[[ ... ]]`.

```suggestion
    local reg="'(.*)'"
    [[ ${__bp_trap_string:-} =~ $reg ]]
    local prior_trap=${BASH_REMATCH[1]-${__bp_trap_string:-}}
```

Reply:
If this works then feel free to submit a PR, not something I have time to test though.

Reply:
OK. The original `shellcheck` warning implied a possibility of reducing the spawning cost of the external command `sed`. The suggestion by `shellcheck` doesn't actually work in this case, so I just showed another way of reducing the cost of `sed`.